orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/StageData.groovy

```diff
@@ -65,9 +66,6 @@ class StageData {
   }
 
   String getAccount() {
-    if (account && credentials && account != credentials) {
```

Comment:
Motivation for removing this sanity check?

Reply:
The clone operation we wanted to perform could be done across Cloud Foundry foundations which would mean we want to perform a clone from different accounts and regions.

we were using `account` as the destination and `credentials` as the the source of the clone. The check prevents the usage of deployment strategy for our clone operation.

Reply:
We are trying to achieve a clone server group stage that can lift an app from one account to another and use typical deployment strategies _a la_ highlander.

`CloneServerGroupStage` has a series of tasks and inheritance that impose these constraints:
1. Determines the server group it is going to clone _from_ with `credentials`
2. Determines which server group to monitor in `WaitForUpInstancesTask` with this (looks at `account` before `credentials`): https://github.com/spinnaker/orca/blob/master/orca-core/src/main/java/com/netflix/spinnaker/orca/clouddriver/utils/CloudProviderAware.java#L51
3. `CloneServerGroupStage` extends from `AbstractDeployStrategyStage`. When a deployment strategy is present on the stage configuration, the strategy handling logic calls this method `StageData#getAccount()`

It is necessary for us to use both `account` to identify the target account and `credentials` to define the source account without requiring significant changes to the way the stage is constructed.

Does this make sense?